File: backend/app/schemas/alert.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from uuid import UUID

class AlertCreate(BaseModel):
    patient_id: UUID
    alert_type: str
    severity: int = Field(ge=1, le=5)
    message: str

class AlertResponse(BaseModel):
    id: UUID
    patient_id: UUID
    alert_type: str
    severity: int
    message: str
    is_acknowledged: bool
    acknowledged_by: Optional[UUID] = None
    acknowledged_at: Optional[datetime] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

class AlertAcknowledge(BaseModel):
    alert_id: UUID

class AlertHistory(BaseModel):
    alerts: List[AlertResponse]
    total: int
    page: int
    limit: int

File: mobile/lib/models/alert.dart
class AppAlert {
  final String id;
  final String alertType;
  final int severity;
  final String message;
  final bool isAcknowledged;
  final DateTime createdAt;
  
  AppAlert({
    required this.id,
    required this.alertType,
    required this.severity,
    required this.message,
    required this.isAcknowledged,
    required this.createdAt,
  });
  
  factory AppAlert.fromJson(Map<String, dynamic> json) {
    return AppAlert(
      id: json['id'] ?? '',
      alertType: json['alert_type'] ?? '',
      severity: json['severity'] ?? 1,
      message: json['message'] ?? '',
      isAcknowledged: json['is_acknowledged'] ?? false,
      createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
    );
  }
}
